Production_scripts/B01_acute_draft.R

```diff
@@ -0,0 +1,332 @@
+    chi = `Pat UPI`,
+    cat = `Patient Category Code`,
+    gpprac = `Practice Location Code`,
+    hbpraccode = `Practice NHS Board Code - current`,
+    selfharm_adm = `Self Harm Related Admission (01)`,
+    sigfac = `Significant Facility Code`,
+    spec = `Specialty Classificat. 1/4/97 Code`,
+    submis_adm = `Substance Misuse Related Admission (01)`,
+    cost_total_net = `Total Net Costs (01)`,
+    location = `Treatment Location Code`,
+    hbtreatcode = `Treatment NHS Board Code - current`,
+    uri = `Unique Record Identifier`,
+    record_keydate1 = `Date of Admission(01)`,
+    record_keydate2 = `Date of Discharge(01)`,
+    dateop1 = `Date of Operation 1 (01)`,
+    dateop2 = `Date of Operation 2 (01)`,
+    dateop3 = `Date of Operation 3 (01)`,
+    dateop4 = `Date of Operation 4 (01)`,
+    dob = `Pat Date Of Birth [C]`,
+    ipdc = `Inpatient Day Case Identifier Code`,
+    cij_ipdc = `CIJ Inpatient Day Case Identifier Code (01)`,
+    lineno = `Line Number (01)`,
+    GLS_record = `GLS Record`
+  )
+
+
+# Data Cleaning ---------------------------------------
+
+acute_clean <- acute_file %>%
+  # Set year variable
+  mutate(
+    year = year,
+    # Set recid as 01B and flag GLS records
+    recid = if_else(GLS_record == "Y", "GLS", "01B"),
+    # Set IDPC marker for the episode
+    ipdc = case_when(
+      ipdc == "IP" ~ "I",
+      ipdc == "DC" ~ "D"
+    ),
+    # Set IDPC marker for the cij
+    cij_ipdc = case_when(
+      cij_ipdc == "IP" ~ "I",
+      cij_ipdc == "DC" ~ "D"
+    )
+  ) %>%
+  # Recode GP practice into 5 digit number
+  # We assume that if it starts with a letter it's an English practice and so recode to 99995.
+  convert_eng_gpprac_to_dummy(gpprac) %>%
+  # Calculate the total length of stay (for the entire episode, not just within the financial year).
+  mutate(
+    stay = difftime(record_keydate2, record_keydate1, units = "days"),
+    # create and populate SMRType
+    SMRType = case_when(
+      recid == "01B" & lineno != 330 ~ if_else(ipdc == "I", "Acute-IP", "Acute-DC")
+      lineno == 330 & ipdc == "I" ~ "GLS-IP",
+      recid == "GLS" ~ "GLS-IP"
+    )
+  ) %>%
+  # Apply new costs for C3 specialty, these are taken from the 2017/18 file
+  fix_c3_costs(year) %>%
+  # initialise monthly cost/beddays variables in a separate data frame for matching
+  convert_monthly_rows_to_vars(uri, costmonthnum, cost_total_net, yearstay) %>%
+  # Add oldtadm as a factor with labels
+  mutate(oldtadm = factor(oldtadm,
+    levels = c(0:8),
+    labels = c(
+      "Deferred admission",
+      "Waiting list/diary/booked",
+      "Repeat admission",
+      "Transfer",
+      "Emergency deliberate self inflicted injury or poisoning",
+      "Emergency road traffic accident",
+      "Emergency home accident (includes accidental poisoning in the home)",
+      "Emergency other injury (includes accidental poisoning other than in the home)",
+      "Emergency other (excluding accidental poisoning)"
+    )
+  ))
```

Comment:
```suggestion
  mutate(oldtadm = factor(oldtadm,
    levels = c(0:8)
  ))
```

as discussed with @Moohan yesterday - remove factor labels as the long strings are too much to try deal with and could break code down the line